includes/admin/tools/class-settings-import.php

```diff
@@ -0,0 +1,681 @@
+				$selected = 'selected';
+			}
+
+			return $selected;
+		}
+
+		/**
+		 * Print the colums from the CSV.
+		 */
+		static function get_columns( $index, $donations, $donors, $forms, $value = false ) {
+			?>
+            <select name="mapto[<?php echo $index; ?>]">
+                <optgroup label="Donations">
+					<?php
+					foreach ( $donations as $option => $option_value ) {
+						$checked = self::selected( $option_value, $value );
+						?>
+                        <option value="<?php echo $option; ?>" <?php echo $checked; ?> ><?php echo $option_value; ?></option>
+						<?php
+					}
+					?>
+                </optgroup>
+
+                <optgroup label="Donors">
+					<?php
+					foreach ( $donors as $option => $option_value ) {
+						$checked = self::selected( $option_value, $value );
+						?>
+                        <option value="<?php echo $option; ?>" <?php echo $checked; ?> ><?php echo $option_value; ?></option>
+						<?php
+					}
+					?>
+                </optgroup>
+
+                <optgroup label="Forms">
+					<?php
+					foreach ( $forms as $option => $option_value ) {
+						$checked = self::selected( $option_value, $value );
+						?>
+                        <option value="<?php echo $option; ?>" <?php echo $checked; ?> ><?php echo $option_value; ?></option>
+						<?php
+					}
+					?>
+                </optgroup>
+
+				<?php
+				do_action( 'give_import_dropdown_option', $index, $donations, $donors, $forms, $value );
+				?>
+            </select>
+			<?php
+		}
+
+		static function get_csv_total( $file_id ) {
+			$total = false;
+			if ( $file_id ) {
+				$file_dir = get_attached_file( $file_id );
+				if ( $file_dir ) {
+					$file = new SplFileObject( $file_dir, 'r' );
+					$file->seek( PHP_INT_MAX );
+					$total = $file->key() + 1;
+				}
+			}
+
+			return $total;
+		}
+
+		static function get_importer( $file_id, $index = 0, $delimiter = ',' ) {
+			$raw_data = false;
+			$file_dir = get_attached_file( $file_id );
+			if ( $file_dir ) {
+				if ( false !== ( $handle = fopen( $file_dir, 'r' ) ) ) {
+					$raw_data = fgetcsv( $handle, $index, $delimiter );
+					// Remove BOM signature from the first item.
+					if ( isset( $raw_data[0] ) ) {
+						$raw_data[0] = self::remove_utf8_bom( $raw_data[0] );
+					}
+				}
+			}
+
+			return $raw_data;
+		}
+
+		/**
+		 * Remove UTF-8 BOM signature.
+		 *
+		 * @param  string $string String to handle.
+		 *
+		 * @return string
+		 */
+		static function remove_utf8_bom( $string ) {
+			if ( 'efbbbf' === substr( bin2hex( $string ), 0, 6 ) ) {
+				$string = substr( $string, 3 );
+			}
+
+			return $string;
+		}
+
+
+		/**
+		 * Is used to show the process when user upload the donor form.
+		 *
+		 * @since 1.2
+		 */
+		static function progress() {
+			$step = Give_Settings_Import::get_step();
+			?>
+            <ol class="give-progress-steps">
+                <li class="<?php echo( 1 === $step ? 'active' : '' ); ?>">Upload CSV file</li>
```

Comment:
@raftaar1191 Is there any reason that these strings are not internationalize

Reply:
Done